doc/graphl/Events.md

```diff
@@ -1,110 +1,150 @@
+
+class GraphQlListener implements EventSubscriberInterface
+{
+    /**
+     * @inheritDoc
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            ExecutorEvents::PRE_EXECUTE => 'onPreExecute', //Pre execute on Query & Mutation
+            ExecutorEvents::POST_EXECUTE => 'onPostExecute' //Post execute on Query & Mutation
+        ];
+    }
+
+    /**
+     * @param ExecutorEvent $event
+     */
+    public function onPreExecute(ExecutorEvent $event)
+    {
+        // do something with the query or schema
+        $query = $event->getQuery();
+        $schema = $event->getSchema();
+    }
+
+    /**
+     * @param ExecutorResultEvent $event
+     */
+    public function onPostExecute(ExecutorResultEvent $event)
+    {
+        // do something with output result
+        $result = $event->getResult();
+    }
+}
+
+```
+
+#### Example 2: Bypass workspace permissions with Query/Mutation build events
+```php
+<?php
+
+namespace AppBundle\EventListener;
+use Pimcore\Bundle\DataHubBundle\Event\GraphQL\MutationEvents;
+use Pimcore\Bundle\DataHubBundle\Event\GraphQL\Model\MutationTypeEvent;
+use Pimcore\Bundle\DataHubBundle\Event\GraphQL\Model\QueryTypeEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class GraphqlListener implements EventSubscriberInterface
+{
+    /**
+     * @inheritDoc
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            MutationEvents::PRE_BUILD => 'onMutationPreBuild',
+            QueryEvents::PRE_BUILD => 'onQueryPreBuild'
+        ];
+    }
+
+    /**
+     * @param ExecutorEvent $event
+     */
+    public function onMutationPreBuild(MutationTypeEvent $event)
+    {
+        $mutationType = $event->getMutationType();
+        $mutationType->setOmitPermissionCheck(true); //omit permission check for mutations
+    }
+
+    /**
+     * @param ExecutorResultEvent $event
+     */
+    public function onQueryPreBuild(QueryTypeEvent $event)
+    {
+        $queryType = $event->getQueryType();
+        $queryType->setOmitPermissionCheck(true); //omit permission check for queries
+    }
+}
+
+```
+
+#### Example 3: Add custom query conditions to object listing
+```php
+<?php
+
+namespace AppBundle\EventListener;
+use Pimcore\Bundle\DataHubBundle\Event\GraphQL\ListingEvents;
+use Pimcore\Bundle\DataHubBundle\Event\GraphQL\Model\ListingEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class GraphqlListener implements EventSubscriberInterface
+{
+    /**
+     * @inheritDoc
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            ListingEvents::PRE_LOAD => 'onListingPreLoad',
+        ];
+    }
+
+    /**
+     * @param ListingEvent $event
+     */
+    public function onListingPreLoad(ListingEvent $event)
+    {
+        $listing = $event->getListing();
+        $args = $event->getArgs();
+
+        if ($args['onlyShowFirstLevelVariants']) {
```

Comment:
where is "onlyShowFirstLevelVariants" set ?

Reply:
This is not a very good example indeed, because it would require defining custom arguments (on own listing types, which is too complicated for just this example query).
So I updated the docs with a more comprehensible example.